src/test_runner.cpp

```diff
@@ -40,35 +72,90 @@ bool test_correct_file(std::filesystem::path path) {
     return true;
 }
 
-bool test_file(std::filesystem::path path) {
+bool test_syntax_file(std::filesystem::path path) {
     if (path.string().ends_with(".fail.lispf")) {
-        return test_fail_file(path);
+        return test_fail_file_syntax(path);
     }
-    return test_correct_file(path);
+    return test_correct_file_syntax(path);
 }
 
-std::vector<std::filesystem::path> get_paths() {
+bool test_semantic_file(std::filesystem::path path) {
+    std::ifstream file(path);
+    std::stringstream buffer;
+    buffer << file.rdbuf();
+    std::string source(buffer.str());
+
+    Reader reader(source);
+
+    auto program = Program::parse(reader.read());
```

Comment:
```suggestion
    Program program;
    try {
        Reader reader(source);
        program = Program::parse(reader.read());
    } catch (SyntaxError const&) {
        return false;
    } catch (EvaluationError const&) {
        return false;
    }
```

вот бы ещё как-то причину фейла сообщать